### PR TITLE
[dy] Add frontend support for managing instances

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add frontend for managing ecs instances from the mage UI. Also made some small backend/infra changes.

# Tests

tested locally and on ecs tasks

Screen Shot 2022-10-13 at 10 20 39 AM

cc: @wangxiaoyou1993

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,10 +5,13 @@ services:
     build:
       context: .
       dockerfile: ./dev.Dockerfile
-    command: "python mage_ai/server/server.py --host ${HOST} --port ${PORT} --project ${PROJECT}"
+    command: "python mage_ai/server/server.py --host ${HOST} --port ${PORT} --project ${PROJECT} --manage-instance ${MANAGE_INSTANCE}"
     environment:
       - AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID
       - AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY
+      - CLUSTER_NAME=$CLUSTER_NAME
```
Comment (wangxiaoyou1993) [inline docker-compose.yml:12]: Let's add prefix to these variables. Otherwise, you can't tell what cluster it is.

File: mage_ai/server/server.py [hunk 2]
```diff
@@ -557,7 +573,11 @@ def start_server(
     host: str = None,
     port: str = None,
     project: str = None,
+    manage: bool = False,
 ):
+    if manage:
+        os.environ[MANAGE_ENV_VAR] = '1'
+
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:580]: Let's not start scheduler for manage instance.

File: mage_ai/server/server.py [hunk 3]
```diff
@@ -390,13 +396,21 @@ def post(self, kernel_id, action_type):
         self.write(r)
         self.finish()
 
+class ApiStatusHandler(BaseHandler):
+    def get(self):
+        status = {
+            'manage': os.getenv(MANAGE_ENV_VAR) == '1',
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:402]: Let's give it a more descriptive name, like "is_instance_manager".

File: scripts/deploy/terraform/aws-dev/main.tf [hunk 4]
```diff
@@ -207,8 +211,8 @@ resource "aws_security_group" "service_security_group" {
     from_port   = 6789
     to_port     = 6789
     protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-    # security_groups = [aws_security_group.load_balancer_security_group.id]
+    cidr_blocks = []
```
Comment (wangxiaoyou1993) [inline scripts/deploy/terraform/aws-dev/main.tf:214]: Let's add your IP address to the security group.

File: mage_ai/server/server.py (hunk 2 above)
Comment (wangxiaoyou1993) [reply mage_ai/server/server.py:580]: Did you address this comment?

Comment (dy46) [reply mage_ai/server/server.py:580]: oops i must have not saved the file

**Comment by dy46** [comment]
@wangxiaoyou1993 updated

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**